### PR TITLE
Added runtime input

### LINKED ISSUE #1260: Updating files to take input at runtime (#1160)
Updating these files :
Babylonian_Method.py
Babylonian_Method.java
Babylonian_Method.go
Exponential_Search.py
Exponential_Search.java
Prime_Factorization.cpp
Prime_Factorization.java
Prime_Factorization.py

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #1260

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Added runtime input

#### Languages Used:
- python
- java
- go
- cpp

#### Files Added:
- Babylonian_Method.py
-  Babylonian_Method.java
- Babylonian_Method.go
- Exponential_Search.py
- Exponential_Search.java
- Prime_Factorization.cpp
- Prime_Factorization.java
- Prime_Factorization.py

### CODE REVIEW

**Comment by somya-kapoor** [comment]
@jainaman224  @manwahmanbir Please review

File: Prime_Factorization/Prime_Factorization.cpp [hunk 1]
```diff
@@ -21,32 +21,33 @@ using namespace std;
 
 void prime_factorize(lint num){
     vector<lint> prime_factors; // To maintain the prime factors of the given number.
-    while(!(num%2)){
+    while(!(num % 2)){
         prime_factors.push_back(2);
-        num/=2;
+        num /= 2;
     }
-    for(lint i=3;i*i<=num;i+=2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space after for

File: Prime_Factorization/Prime_Factorization.cpp [hunk 2]
```diff
@@ -21,32 +21,33 @@ using namespace std;
 
 void prime_factorize(lint num){
     vector<lint> prime_factors; // To maintain the prime factors of the given number.
-    while(!(num%2)){
+    while(!(num % 2)){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space after while

File: Prime_Factorization/Prime_Factorization.cpp [hunk 3]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
         prime_factors.push_back(2);
-        num/=2;
+        num /= 2;
     }
-    for(lint i=3;i*i<=num;i+=2){
-        while(!(num%i)){
+    for(lint i = 3; i * i <= num; i+=2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space after for and +=

File: Prime_Factorization/Prime_Factorization.cpp [hunk 4]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
+        while(!(num % i)){
             prime_factors.push_back(i);
-            num/=i;
+            num /= i;
         }
     }
-    if(num>2){
+    if(num > 2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space after if

File: Prime_Factorization/Prime_Factorization.cpp [hunk 5]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
         prime_factors.push_back(num);
     }
     // Printing the prime factors.
-    for(lint i=0;i<prime_factors.size();i++){
-        cout<<prime_factors[i];
-        if(i != prime_factors.size()-1){
-            cout<<" * ";
+    for(lint i = 0; i < prime_factors.size(); i++){
+        cout << prime_factors[i];
+        if(i != prime_factors.size() - 1){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space after if

**Comment by somya-kapoor** [comment]
@MastersAbh done please review

**Comment by somya-kapoor** [comment]
@jainaman224 @marwahmanbir Please review

File: Exponential_Search/Exponential_Search.java [hunk 6]
```diff
@@ -22,16 +22,28 @@ static int expoSearch(int input[], int n, int x)
...
     { 
-	// Sample Input
-        int arr[] = {2, 3, 14, 20, 40}; 
-	int x = 20; 
+	
+	int n;
+	Scanner s = new Scanner(System.in);
+	n = s.nextInt();
+        int arr[] = new int[n];
+	for(int i = 0; i < n; i++)
```
Comment (MastersAbh) [inline Exponential_Search/Exponential_Search.java]: space after for

File: Prime_Factorization/Prime_Factorization.cpp [hunk 7]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
     }
     // Printing the prime factors.
-    for(lint i=0;i<prime_factors.size();i++){
-        cout<<prime_factors[i];
-        if(i != prime_factors.size()-1){
-            cout<<" * ";
+    for (lint i = 0; i < prime_factors.size(); i++){
+        cout << prime_factors[i];
+        if (i != prime_factors.size() - 1){
+            cout << " * ";
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: remove spaces around <<

File: Prime_Factorization/Prime_Factorization.cpp [hunk 8]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
     }
 }
 
 int main()
 {
-    lint num = 100;
-    cout<<num<<" = ";
+    lint num;
+    cin >> num
+    cout << num << " = ";
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: remove spaces around << and >>

File: Prime_Factorization/Prime_Factorization.java [hunk 9]
```diff
@@ -26,31 +27,33 @@
 
     public static void prime_factorize(long num){
         List<Long> prime_factors = new ArrayList<Long>(); // To maintain the prime factors of the given number.
-        while(num%2 == 0){
+        while(num % 2 == 0){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space after while

File: Prime_Factorization/Prime_Factorization.java [hunk 10]
```diff
@@ -26,31 +27,33 @@
...
             prime_factors.add((long)2);
-            num/=2;
+            num /= 2;
         }
-        for(long i=3;i*i<=num;i+=2){
-            while(num%i == 0){
+        for(long i = 3; i * i <= num; i+=2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space around +=

File: Prime_Factorization/Prime_Factorization.java [hunk 11]
```diff
@@ -26,31 +27,33 @@
...
+            while(num % i == 0){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space after while , {

File: Prime_Factorization/Prime_Factorization.java [hunk 12]
```diff
@@ -26,31 +27,33 @@
...
                 prime_factors.add(i);
-                num/=i;
+                num /= i;
             }
         }
-        if(num>2){
+        if(num > 2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space after if

File: Prime_Factorization/Prime_Factorization.java [hunk 13]
```diff
@@ -26,31 +27,33 @@
...
             prime_factors.add(num);
         }
         // Printing the prime factors.
-        for(int i=0;i<prime_factors.size();i++){
+        for(int i = 0; i < prime_factors.size(); i++){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space after for

File: Prime_Factorization/Prime_Factorization.java [hunk 14]
```diff
@@ -26,31 +27,33 @@
...
             System.out.print(prime_factors.get(i));
-            if(i != prime_factors.size()-1){
+            if(i != prime_factors.size() - 1){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: space after if

File: Prime_Factorization/Prime_Factorization.java [hunk 15]
```diff
@@ -26,31 +27,33 @@
...
             }
         }
     }
 
     public static void main(String[] args) {
-        long num = 100;
-        System.out.print(num+" = ");
+        long num ;
+        Scanner s = new Scanner(System.in);
+	    num = s.nextLong();
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: please fix indentation

File: Prime_Factorization/Prime_Factorization.py [hunk 16]
```diff
@@ -15,24 +15,24 @@
     After the loop, if the value of num still tends to be greater than 2, it means,
     it is a prime factor in itself. Thus, we add to it's prime factors.
 '''
-def prime_factorize(num):
+def prime_factorize(num) :
     prime_factors = []  # To maintain the prime factors of the given number.
-    while(not (num%2)):
+    while (not (num % 2)) :
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.py]: please dont add space before : in python

File: Prime_Factorization/Prime_Factorization.py [hunk 17]
```diff
@@ -15,24 +15,24 @@
...
-    for i in range(3,int(math.sqrt(num))+1):
-        while(not (num%i)):
+        num /= 2
+    for i in range(3, int(math.sqrt(num)) + 1) :
+        while (not (num % i)):
             prime_factors.append(i)
-            num/=i
-    if num>2:
+            num /= i
+    if num > 2 :
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.py]: please dont add space before : in python

File: Prime_Factorization/Prime_Factorization.py [hunk 18]
```diff
@@ -15,24 +15,24 @@
...
-        if not (i == len(prime_factors)-1):
-            print(" * ",end='')
-num = 100
-print(num," = ",end='')
+    for i in range(0, len(prime_factors)) :
+        print(prime_factors[i], end = '')
+        if not (i == len(prime_factors) - 1) :
+            print(" * ", end = '')
+num = int(input())
+print(num, " = ", end ='')
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.py]: space after =

File: Prime_Factorization/Prime_Factorization.py [hunk 19]
```diff
@@ -15,24 +15,24 @@
...
     # Printing the prime factors.
-    for i in range(0,len(prime_factors)):
-        print(prime_factors[i],end='')
-        if not (i == len(prime_factors)-1):
-            print(" * ",end='')
-num = 100
-print(num," = ",end='')
+    for i in range(0, len(prime_factors)) :
+        print(prime_factors[i], end = '')
+        if not (i == len(prime_factors) - 1) :
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.py]: please dont add space before : in python

File: Prime_Factorization/Prime_Factorization.py [hunk 20]
```diff
@@ -15,24 +15,24 @@
...
+    if num > 2 :
         prime_factors.append(num)
     # Printing the prime factors.
-    for i in range(0,len(prime_factors)):
-        print(prime_factors[i],end='')
-        if not (i == len(prime_factors)-1):
-            print(" * ",end='')
-num = 100
-print(num," = ",end='')
+    for i in range(0, len(prime_factors)) :
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.py]: please dont add space before : in python

File: Exponential_Search/Exponential_Search.java [hunk 21]
```diff
@@ -1,4 +1,4 @@
-
+import java.util.Scanner;
```
Comment (MastersAbh) [inline Exponential_Search/Exponential_Search.java]: Hi, java.util.* is already imported. Can you remove one of those?

File: Exponential_Search/Exponential_Search.java (hunk 6 above)
Comment (MastersAbh) [reply Exponential_Search/Exponential_Search.java]: Still not done

File: Exponential_Search/Exponential_Search.py [hunk 22]
```diff
@@ -36,15 +36,23 @@ def exponentialSearch(arr, n, x):
...
-final = exponentialSearch(input, n, x) 
+n = int(input())
+arr = []
+for i in range(n):
+    x = int(input())
+    arr.append(x) 
+x = int(input())       
+final = exponentialSearch(arr, n, x) 
 if final == -1: 
     print "Element not found in thye array"
```
Comment (MastersAbh) [inline Exponential_Search/Exponential_Search.py]: typo in print statement (thye)

File: Prime_Factorization/Prime_Factorization.cpp [hunk 23]
```diff
@@ -21,32 +21,33 @@ using namespace std;
 
 void prime_factorize(lint num){
     vector<lint> prime_factors; // To maintain the prime factors of the given number.
-    while(!(num%2)){
+    while (!(num % 2)){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space before {

File: Prime_Factorization/Prime_Factorization.cpp [hunk 24]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
         prime_factors.push_back(2);
-        num/=2;
+        num /= 2;
     }
-    for(lint i=3;i*i<=num;i+=2){
-        while(!(num%i)){
+    for (lint i = 3; i * i <= num; i += 2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space before {

File: Prime_Factorization/Prime_Factorization.cpp [hunk 25]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
+        while(!(num % i)){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space before {

File: Prime_Factorization/Prime_Factorization.cpp [hunk 26]
```diff
@@ -21,32 +21,33 @@ using namespace std;
...
             prime_factors.push_back(i);
-            num/=i;
+            num /= i;
         }
     }
-    if(num>2){
+    if (num > 2){
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.cpp]: space before {

File: Prime_Factorization/Prime_Factorization.cpp (hunk 7 above)
Comment (MastersAbh) [reply Prime_Factorization/Prime_Factorization.cpp]: space before { at all places

File: Prime_Factorization/Prime_Factorization.java [hunk 27]
```diff
@@ -5,7 +5,8 @@
     i.e. denote a number as a product of different prime factors.
 
 */
-
+    
+import java.util.Scanner;
```
Comment (MastersAbh) [inline Prime_Factorization/Prime_Factorization.java]: util.* is already imported.

File: Prime_Factorization/Prime_Factorization.java (hunk 9 above)
Comment (MastersAbh) [reply Prime_Factorization/Prime_Factorization.java]: space before {

File: Prime_Factorization/Prime_Factorization.java (hunk 13 above)
Comment (MastersAbh) [reply Prime_Factorization/Prime_Factorization.java]: please do this

**Comment by somya-kapoor** [comment]
@MastersAbh Done please review

**Pull request merged**